modules/402-ingress-nginx/images/kruise/Dockerfile

```diff
@@ -1,24 +1,30 @@
-ARG BASE_ALPINE
+ARG BASE_DISTROLESS
 ARG BASE_GOLANG_19_BULLSEYE
 
 FROM $BASE_GOLANG_19_BULLSEYE as builder
 ARG KRUISE_CONTROLLER_VERSION=1.4.0
+ARG SOURCE_REPO
+ARG GOPROXY
+
+ENV GOPROXY=${GOPROXY} \
+    SOURCE_REPO=${SOURCE_REPO} \
+    CGO_ENABLED=0 \
+    GOOS=linux \
+    GOARCH=amd64
 
 COPY patches/disable-controllers.patch /
 COPY patches/disable-jobs.patch /
 COPY patches/stick-to-maxunavailable.patch /
 
-RUN git clone --depth 1 --branch v${KRUISE_CONTROLLER_VERSION} https://github.com/openkruise/kruise.git && \
+RUN git -c advice.detachedHead=false clone --depth 1 --branch v${KRUISE_CONTROLLER_VERSION} ${SOURCE_REPO}/openkruise/kruise.git && \
```

Comment:
why do we need this option here ?

Reply:
-c advice.detachedHead=false ?

Reply:
git exits with non-zero code without it if you hit detached state after clone without this iirc